### PR TITLE
Javascript - javascript1 - week2

### PR DESCRIPTION
**Pull request by alina-kho**
Homework for the second week of the first JS module

### CODE REVIEW

File: javascript/javascript1/week2/candy-help.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+console.log(boughtCandyPrices);
+
+function canBuyMoreCandy(candiesBought, budget) {
+  let sum = 0;
+  for (i = 0; i < candiesBought.length; i++) {
+    sum += candiesBought[i];
+  }
+  if (sum < amountToSpend) {
+    console.log("You can buy more, so please do!");
+    return true;
```
Comment (vulk29) [inline javascript/javascript1/week2/candy-help.js]: im not sure you need the returns here.. do you?

File: javascript/javascript1/week2/events.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  let d = new Date().getDay(); //today's index
+  let eventIndex = (days % 7) + d; // the remainder = how far the event's week day from today's weekday (if we consider just a week) E.g. remainder = 0 - the event is on the same weekday as today's
+  var eventDay; // introducing a variable for a further use in an if-statement
+  if (eventIndex >= 7) {
+    eventDay = daysOfWeek[eventIndex % 7]; // the array has indexes only from 0 to 6 - sometimes the eventIndex is higher, so we need to get a remainder once again but it's already a sum of indexes
+  } else {
+    eventDay = daysOfWeek[eventIndex];
+  }
+  return eventDay;
+}
```
Comment (vulk29) [inline javascript/javascript1/week2/events.js:20]: this works, but a simpler way would be to put them together so you would have directly. Whenever possible, aim to make your code shorter :)
eventday=daysOfWeek[(days + d) % 7];

File: javascript/javascript1/week2/fullname.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+function getFullname(firstname, lastname, useFormalName, sex) {
+  if (firstname === "" || lastname === "") {
+    alert("Please enter a full name!"); //a bit of validation, so the data is provided
+  } else if (useFormalName === true) {
```
Comment (vulk29) [inline javascript/javascript1/week2/fullname.js:4]: The alert can also just be a message  - i get an error with the current implementation. 
so e.g. return "Please enter a full name"

File: javascript/javascript1/week2/fullname.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    } else {
+      return firstname + " " + lastname; // so we include those who prefer not to mention info about their sex
+    }
+  }
+}
+
+//Checking if it works :)
+//getFullname("", "Black", true, "male"); (commented it so you don't get an alert right away)
+getFullname("Benjamin", "Hughes", true, "male");
+getFullname("Benjamin", "Hughes", false, "male");
```
Comment (vulk29) [inline javascript/javascript1/week2/fullname.js:25]: if you wrap a console.log around the getFullname function, you will also get a print out :)
like console.log(getFullname("Benjamin", "Hughes", true, "male");

File: javascript/javascript1/week2/s-manager.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+addStudentToClass("John"); //works
+addStudentToClass("Alice"); //works
+addStudentToClass("Mark");
+addStudentToClass("Jane");
+addStudentToClass("Peter");
+addStudentToClass("Michael");
+addStudentToClass("Paula"); //works
+addStudentToClass("Queen"); // works
+
+console.log(getNumberOfStudents());
```
Comment (vulk29) [inline javascript/javascript1/week2/s-manager.js:39]: great job here! these nested if else statements are always tricky to manage!

File: javascript/javascript1/week2/w-wear.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  } else if (temp <= 25) {
+    return "jeans or shorts and t-shirt";
+  } else if (temp > 25) {
+    return "short and t-shirt";
+  }
+}
+
+//Test
+let outfit = getYourOutfit(21);
+console.log(outfit);
```
Comment (vulk29) [inline javascript/javascript1/week2/w-wear.js:23]: hahah. i love your -30 statement. Good job!

**Review by vulk29 — CHANGES_REQUESTED** [review]

File: javascript/javascript1/week2/fullname.js (hunk 3 above)
Comment (alina-kho) [reply javascript/javascript1/week2/fullname.js:4]: I will fix it :) Thank you!

File: javascript/javascript1/week2/events.js (hunk 2 above)
Comment (alina-kho) [reply javascript/javascript1/week2/events.js:20]: You are right! I have overthought a bit, gonna fix it

**Pull request merged**